Source/Web View/WebViewController.swift

```diff
@@ -292,40 +296,47 @@ extension WebViewController {
      :param: url The URL used to load the web view.
     */
     final public func loadURL(url: NSURL) {
-        webView.stopLoading()
+        if self.dataTask?.state == .Running { // allow running task to finish
+            return
+        }
         hybridAPI = nil
         firstLoadCycleCompleted = false
 
         self.url = url
         let request = requestWithURL(url)
 
-        let dataTask: NSURLSessionDataTask = self.urlSession.dataTaskWithRequest(request) { (data, response, error) -> Void in
-            if let urlResponse = response as? NSHTTPURLResponse {
-                if (urlResponse.statusCode >= 400) || (error != nil) {
-                    // handle error condition
-                    var httpError = error
-                    if httpError == nil {
-                        httpError = NSError(domain: "WebViewController", code: urlResponse.statusCode, userInfo: ["response" : urlResponse, NSLocalizedDescriptionKey : "HTTP Response Status \(urlResponse.statusCode)"])
+        self.dataTask = self.urlSession.dataTaskWithRequest(request) { (data, response, error) -> Void in
+            dispatch_async(dispatch_get_main_queue(), { () -> Void in
+                if error?.code == -999 || self.dataTask?.state != .Completed { // cancelled
+                    return
```

Comment:
also should not need this return if we call `self.dataTask?.cancel()` because canceling the data task will prevent the completion handler from being called